src/Amazon.Common.DotNetCli.Tools/Commands/BaseCommand.cs

```diff
@@ -113,6 +113,23 @@ protected static IList<CommandOption> BuildLineOptions(params IList<CommandOptio
             return list;
         }
 
+        /// <summary>
+        /// Used to combine the command specific command options into one list.
+        /// </summary>
+        /// <param name="optionCollections"></param>
+        /// <returns></returns>
+        protected static IList<CommandOption> MergeCommandOptions(params IList<CommandOption>[] optionCollections)
```

Comment:
I don't want another method that does 9X% the same as `BuildLineOptions`. Lets modify `BuildLineOptions` to only add things to the union list if it isn't already there. Something like
```
protected static IList<CommandOption> BuildLineOptions(params IList<CommandOption>[] optionCollections)
{
    var list = new List<CommandOption>();
    list.AddRange(CommonOptions);
    if(optionCollections != null)
    {
        foreach (var options in optionCollections)
        {
            foreach(var option in options)
            {
                if (!list.Any(x => string.Equals(x.Switch, option.Switch)))
                {
                    list.Add(option);
                }
            }
        }
    }

    return list;
}
```

Then be sure in the `PushDockerImageCommand` we pass up `BasePushDockerImageCommand<LambdaToolsDefaults>.CommandOptions` instead of `CommonOptions`.

Reply:
Done. Please review.